src/cloud/functions/leaderboards/around-me/leaderboard/run.csx

```diff
@@ -0,0 +1,240 @@
+private const int radiusOfLeaderboard = 5;
+private static bool runOnce = true;
+private static DocumentClient client;
+
+public static async Task<HttpResponseMessage> Run(HttpRequestMessage req, string leaderboard, string playerId, TraceWriter log)
+{
+    // Run initialization only once.
+    // Remarks: This initialization will run once on every instance and on every recompile of this function
+    if (runOnce)
+    {
+        log.Info("Running initialization");
+
+        if (string.IsNullOrWhiteSpace(baseArchitectureVersion) ||
+            !baseArchitectureVersion.StartsWith(requiredBaseArchitectureVersion)) log.Error($"The base architecture version doesn't match the expected version {requiredBaseArchitectureVersion}");
+
+        // Check required application settings
+        if (string.IsNullOrWhiteSpace(db)) log.Error("COSMOSDB_DBNAME settings wasn't provided");
+        if (string.IsNullOrWhiteSpace(endpoint)) log.Error("COSMOSDB_ENDPOINT settings wasn't provided");
+        if (string.IsNullOrWhiteSpace(key)) log.Error("COSMOSDB_PRIMARY_MASTER_KEY settings wasn't provided");
+
+        // Create Cosmos DB Client from settings
+        client = new DocumentClient(new Uri(endpoint), key);
+
+        // Create Database and Collection in Cosmos DB Account if they don't exist
+        await client.CreateDatabaseIfNotExistsAsync(new Database { Id = db });
+        await client.CreateDocumentCollectionIfNotExistsAsync(
+            UriFactory.CreateDatabaseUri(db),
+            new DocumentCollection { Id = collection });
+
+        runOnce = false;
+
+        log.Info("Initialization done!");
+    }
+
+    if (string.IsNullOrWhiteSpace(leaderboard) || string.IsNullOrWhiteSpace(playerId))
+        return req.CreateResponse(HttpStatusCode.BadRequest, "Both playerId and leaderboard should be provided in querystring, format is leaderboard/playerId");
+
+    try
+    {
+
+        var scores = client.CreateDocumentQuery<ScoreItem>(
+            UriFactory.CreateDocumentCollectionUri(db, collection), new FeedOptions { EnableCrossPartitionQuery = true });
+
+        // Get score for the current player by playerId        
+        var query =
+            (from s in scores
+             where s.PlayerId == playerId && s.Leaderboard == leaderboard
+             select s);
+
+        var player = query.ToList();
+        if (player.Count < 1)
+            return req.CreateResponse(HttpStatusCode.BadRequest, $"Player with Id = {playerId} in Leaderboard = {leaderboard} does not exist in the database");
+
+        if (player.Count > 1)
+            return req.CreateResponse(HttpStatusCode.BadRequest, $"There are more than one player with Id = {playerId} in Leaderboard = {leaderboard}. This is DB confuguration error - this combination should be unique");
+
+        var currentPlayer = player.First();
+
+        // Calculate global rank for the current player by playerId 
+        int globalRank;
+
+        query = from c in scores
```

Comment:
Re-using the same query variable is doable, but causes readability issues. Please create new variables with meaningful names if they no longer mean the same thing within a single method.